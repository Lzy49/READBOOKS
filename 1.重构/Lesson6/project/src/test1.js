import day from "dayjs";
export function printOwing(invoice) {
  printBanner();
  const outstanding = calclateOutstanding(invoice);
  recordDueDate(invoice);
  printDetails(invoice, outstanding);
  return outstanding;
}
function recordDueDate(invoice) {
  const today = day();
  invoice.dueDate = today.add(30, "d");
}
function calclateOutstanding(invoice) {
  let result = 0;
  for (const o of invoice.orders) {
    result += o.amount;
  }
  return result;
}
function printBanner() {
  console.log("----------------");
  console.log("- Custome Owes -");
  console.log("----------------");
}
function printDetails(invoice, outstanding) {
  console.log(`name:${invoice.customer}`);
  console.log(`amount:${outstanding}`);
  console.log(`due:${invoice.dueDate.format("YYYY/MM/DD")}`);
}
