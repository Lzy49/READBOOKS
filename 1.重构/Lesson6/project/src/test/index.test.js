import { describe, it, expect } from "vitest";
import { printOwing } from "../test1";
describe("happy path", () => {
  it("test text", () => {
    const invoice = {
      customer:'Justin',
      orders:[
        {
          amount:10
        },
        {
          amount:11
        },
        {
          amount:20
        }
      ]
    }
    expect(printOwing(invoice)).toBe(41)
  });
});
