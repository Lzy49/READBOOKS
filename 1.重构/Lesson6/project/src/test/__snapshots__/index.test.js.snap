// Vitest Snapshot v1

exports[`happy path > test HTML 1`] = `
"<h1>Statement for BigCo</h1>
<table>
<tr><th>player</th><th>amount</th><th>audience</th></tr><tr><td>Hamlet</td><td>$650.00</td><td>55 seats</td></tr></td>
<tr><td>As You like It</td><td>$490.00</td><td>35 seats</td></tr></td>
<tr><td>Othello</td><td>$500.00</td><td>40 seats</td></tr></td>
</table> 
<p>Amount owed is $1,640.00</p>
<p>You earned 47 credits</p>
"
`;

exports[`happy path > test text 1`] = `
"Statement for BigCo
 Hamlet:$650.00(55 seats)
 As You like It:$490.00(35 seats)
 Othello:$500.00(40 seats)
Amount owed is $1,640.00
You earned 47 credits
"
`;
